src/source.py

```diff
@@ -16,4 +16,82 @@ def d2YdtdE_0_func_doublegauss(t, E, N, mu_t, sig_t, mu_E, sig_E):
 
 def dYdt_0_func_gauss(t, E, N, mu_t, sig_t):
     dYdt_0 = N/np.sqrt(2*np.pi)/sig_t * np.exp(-0.5 * ((t-mu_t)/sig_t)**2)
-    return dYdt_0
\ No newline at end of file
+    return dYdt_0
+
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Feb 20 09:27:10 2023
+
+@author: blr
+"""
+import numpy as np
+import scipy
+
+##############################################################################
+########################### MODEL FUNCTIONS ##################################
+##############################################################################
+
+def d2YdtdE_0_func_doublegauss(t, E, N, mu_t, sig_t, mu_E, sig_E):
+    d2YdtdE_0 = N/2/np.pi/sig_t/sig_E * np.exp(-0.5 * (((t-mu_t)/sig_t)**2 + ((E-mu_E)/sig_E)**2))
+    return d2YdtdE_0
+
+def dYdt_0_func_gauss(t, E, N, mu_t, sig_t):
+    dYdt_0 = N/np.sqrt(2*np.pi)/sig_t * np.exp(-0.5 * ((t-mu_t)/sig_t)**2)
+    return dYdt_0
+
+'''
+return energy spectrum of product from fusion reactions in plasma w/ ion temperature Ti [keV]
+and yield Y
+'''
+def ballabio_spectrum(reaction, Ti, Y):
+    if reaction == "DD":
+        a1E = 4.69515
+        a2E = -0.040729
+        a3E = 0.47
+        a4E = 0.81844
+        a1w = 1.7013 * 1e-3
+        a2w = 0.16888
+        a3w = 0.49
+        a4w = 7.9460 * 1e-4
+        E0 = 2.4495 # MeV
+        w0 = 82.542 # kev^(1/2)
+    
+        # interpolation formulas for deviation from standard results
+        deltaE = (a1E/(1 + a2E * (Ti**a3E))) * (Ti**(2./3)) + a4E*Ti # keV
```

Comment:
Given that the stuff in here is reused for several different reactions, I would turn it into a separate function in the module that you call which will help with reducing clutter and aids with testing/ensuring there aren't transcription errors if the function is expanded in the future